From 10b974fde9049e46436f885d155be7bc937d47c0 Mon Sep 17 00:00:00 2001
From: John Kacur <jkacur@redhat.com>
Date: Tue, 8 Dec 2015 02:13:03 +0100
Subject: [PATCH 4/4] trace-cmd: trace-util.c Prevent resource leak by closing
 fp

Close filepointer fp to prevent resource leak on error path
before return of NULL

Fix a few spacing problems (extraneous tabs and spaces) in the function

Signed-off-by: John Kacur <jkacur@redhat.com>
---
 trace-util.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/trace-util.c b/trace-util.c
index f71b5ad0f203..600aa7f1b34b 100644
--- a/trace-util.c
+++ b/trace-util.c
@@ -650,7 +650,7 @@ char *tracecmd_find_tracing_dir(void)
 	char type[100];
 	int use_debug = 0;
 	FILE *fp;
-	
+
 	if ((fp = fopen("/proc/mounts","r")) == NULL) {
 		warning("Can't open /proc/mounts for read");
 		return NULL;
@@ -664,8 +664,10 @@ char *tracecmd_find_tracing_dir(void)
 			break;
 		if (!debug_str && strcmp(type, "debugfs") == 0) {
 			debug_str = strdup(fspath);
-			if (!debug_str)
+			if (!debug_str) {
+				fclose(fp);
 				return NULL;
+			}
 		}
 	}
 	fclose(fp);
@@ -698,7 +700,7 @@ char *tracecmd_find_tracing_dir(void)
 		tracing_dir = strdup(fspath);
 		if (!tracing_dir)
 			return NULL;
-	}		
+	}
 
 	return tracing_dir;
 }
-- 
2.4.3

